app/code/Magento/Catalog/Model/ResourceModel/Category/Collection.php

```diff
@@ -125,6 +131,7 @@ public function __construct(
             \Magento\Framework\App\ObjectManager::getInstance()->get(ScopeConfigInterface::class);
         $this->catalogProductVisibility = $catalogProductVisibility ?:
             \Magento\Framework\App\ObjectManager::getInstance()->get(Visibility::class);
+        $this->tableMaintainer = \Magento\Framework\App\ObjectManager::getInstance()->get(TableMaintainer::class);
```

Comment:
Out of curiosity, why is the `TableMaintainer` dependency not added as a constructor argument in the same way as the `Visibility` instance?